paasta_tools/setup_marathon_job.py

```diff
@@ -577,19 +615,21 @@ def log_deploy_error(errormsg: str, level: str='event') -> None:
             scaling_app_at_risk_tasks = list(task_dict['at_risk'])
 
             tasks_to_move_draining = min(len(scaling_app_draining_tasks), num_tasks_to_scale)
-            old_app_draining_tasks[new_app.id] = set(scaling_app_draining_tasks[:tasks_to_move_draining])
+            old_app_draining_tasks[(new_app.id, new_client)] = set(scaling_app_draining_tasks[:tasks_to_move_draining])
```

Comment:
Isn't this code assuming that the old apps only exist on the new_client? Isn't that assumption incorrect?

Reply:
No, it's assuming that the new app only exists on the new client.

This chunk of code is putting new tasks in the various categories for old tasks if we need to scale down the current app; that way, later code will drain and kill them.